### ISSUE TITLE
No credits on Travis CI for build job

### ISSUE THREAD

**Issue by sidey79** [fhem/fhem-docker#28]
**Describe the bug**
Travis CI has migrated to credit based service.
May there is a chance of getting credits from travis for that open source project, but on the other hand the build process can be migrated to github actions which integrates better into github overall.

![image](https://user-images.githubusercontent.com/7968127/103210662-a01fc480-4906-11eb-8a8d-30c11e8728a5.png)

**To Reproduce**
1. Go to https://travis-ci.com/github/fhem/fhem-docker
2. Look at the banner and the build process

**Additional context**
![image](https://user-images.githubusercontent.com/7968127/103211536-c21a4680-4908-11eb-95ca-3c27f2dfb4f7.png)

**Comment by sidey79** [comment]
Tryed to build in GitHub actions.

https://github.com/sidey79/fhem-docker/actions/runs/448781424

Some more work needs to be done.
If seems that it aborts at some npm Installation, not sure for what this is needed on a Perl project :)

**Comment by jpawlowski** [comment]
Thanks for helping out here.
I had addressed this issue by reducing repetitive build tasks but could not follow up on it as of today. My expectation is that it would allow limited builds on Travis. However, some crypto Perl modules are unstable again which were blocking successful builds, resulting in a waste of build time.

> Tryed to build in GitHub actions.
> 
> https://github.com/sidey79/fhem-docker/actions/runs/448781424

I have added my comments to PR #29.

> If seems that it aborts at some npm Installation, not sure for what this is needed on a Perl project :)

There are popular FHEM modules that integrate with NPM backend modules, in particular for voice command assistant integrations.

**Comment by sidey79** [comment]
Travis doesn't grant new credits by default. May we can request for oss credits, but i think building on github actions isn't a bad idea overall.

Build time long during testing and wastes time.
I did a deep dive into this and figured out that the build cache is invalidated at a very early layer.

I already worked on a fix to use build cache as long as possible. This will speed up the build time. I only need to get all platforms working, which is some of a nightmare, because some packages have different versions per platform.
